<sdf version="1.6">
    <world name="pupper_world">
        <include>
            <uri>model://sun</uri>
        </include>

        <include>
			<uri>model://ground_plane</uri>
		</include>
        <gravity>
            0 0 -9.81
        </gravity>
        <physics name="ode_100iters" type="ode" default="true">
            <type>ode</type>
            <max_step_size>0.0005</max_step_size>
            <real_time_update_rate>1000</real_time_update_rate>
            <max_contacts>4</max_contacts> 
            <ode>

                <solver>
                    <type>quick</type>
                    <iters>50</iters>
                    <sor>0.4</sor>
                </solver>

                <constraints>
                    <cfm>0.0</cfm>
                    <contact_max_correcting_vel>1000</contact_max_correcting_vel>
                    <contact_surface_layer>0.001</contact_surface_layer>
                </constraints>

            </ode>
        </physics>
    </world>
</sdf>
